src/ripple/core/SQLInterface_nodes.h

```diff
@@ -0,0 +1,201 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_SQLINTERGACE_NODES_H_INCLUDED
+#define RIPPLE_CORE_SQLINTERGACE_NODES_H_INCLUDED
+
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/core/Config.h>
+#include <ripple/core/SQLInterface.h>
+#include <ripple/overlay/PeerReservationTable.h>
+#include <ripple/peerfinder/impl/Store.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+
+enum class TableType { Ledgers, Transactions, AccountTransactions};
+
+std::tuple<bool, std::unique_ptr<DatabaseCon>, std::unique_ptr<DatabaseCon>>
```

Comment:
In general, I dislike `pair` and `tuple` if any of the slots have the same type. It's too easy to mix up what's what. Use a struct return type here?

Reply:
Done.